### PR TITLE
fix(theme-classic) extract HomeBreadcrumbItem + fix swizzle bugs

### PR DESCRIPTION
**Pull request by 3v0k4**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

See https://github.com/facebook/docusaurus/issues/6953#issuecomment-1346218020

### 1.

`yarn test` (in the repo root) is green

### 2.

I changed `website/package.json`:

```diff
-     "@docusaurus/theme-classic": "^3.0.0-alpha.0",
+     "@docusaurus/theme-classic": "file:../packages/docusaurus-theme-classic",
```

and then

```
cd website
yarn install
npm run swizzle @docusaurus/theme-classic DocBreadcrumbs/HomeBreadcrumbItem -- --eject
```

Notice that swizzling produces the following warning:

```
[WARNING]
Swizzle action eject is unsafe to perform on DocBreadcrumbs/HomeBreadcrumbItem.
It is more likely to be affected by breaking changes in the future
If you want to swizzle it, use the `--danger` flag, or confirm that you understand the risks.

? Do you really want to swizzle this unsafe internal component? › - Use arrow-keys. Return to submit.
    NO: cancel and stay safe
❯   YES: I know what I am doing!
    [Exit]
```

I also changed the icon in the .tsx

```
<Link ...>
  <IconHome className={styles.breadcrumbHomeIcon} />
  HOME SWEET HOME <-----
</Link>
```

and .css

```
.breadcrumbHomeIcon {
  position: relative;
  top: 1px;
  vertical-align: top;
  height: 1.1rem;
  width: 1.1rem;
  color: red; <-----
}
```

which resulted in

Red home icon and HOME SWEET HOME on the right

### Test links

Deploy preview: https://deploy-preview-_____--docusaurus-2.netlify.app/

## Related issues/PRs

See Motivation above.

### CODE REVIEW

**Comment by 3v0k4** [comment]
Is `styles.breadcrumbsItemLink` unused code that we can remove?

File: packages/docusaurus-theme-classic/src/theme-classic.d.ts [hunk 1]
```diff
@@ -1510,3 +1510,7 @@ declare module '@theme/prism-include-languages' {
     PrismObject: typeof PrismNamespace,
   ): void;
 }
+
+declare module '@theme/DocBreadcrumbs/HomeBreadcrumbItem' {
```
Comment (slorber) [inline packages/docusaurus-theme-classic/src/theme-classic.d.ts:1514]: Considering we are going to have multiple items I'd prefer to have something like `@theme/DocBreadcrumbs/Items/Home`

File: packages/docusaurus-theme-classic/src/theme/DocBreadcrumbs/HomeBreadcrumbItem/index.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  return (
+    <li className="breadcrumbs__item">
+      <Link
+        aria-label={translate({
+          id: 'theme.docs.breadcrumbs.home',
+          message: 'Home page',
+          description: 'The ARIA label for the home page in the breadcrumbs',
+        })}
+        className={clsx('breadcrumbs__link', styles.breadcrumbsItemLink)}
```
Comment (slorber) [inline packages/docusaurus-theme-classic/src/theme/DocBreadcrumbs/HomeBreadcrumbItem/index.tsx:28]: indeed it looks like a dead style 👍 

```suggestion
        className="breadcrumbs__link"
```

**Review by slorber — CHANGES_REQUESTED** [review]
Thanks 👍 

See PR comments

CleanShot 2022-12-16 at 16 55 05@2x

No need to, monorepos hoist and use local packages ;)

**Comment by 3v0k4** [comment]
> See PR comments

✅

> No need to, monorepos hoist and use local packages ;)

TIL 🙏

**Review by slorber — APPROVED** [review]
Thanks 👍 LGTM

**Comment by slorber** [comment]
hmmm, little CI failure that I will have to check, not your fault :)

**Pull request merged**